### ISSUE TITLE
Setting --workDir doesn't affect config.json search path

### ISSUE THREAD

**Issue by bcat** [PufferPanel/PufferPanel#1150]
Overall, the `workDir` command line argument mostly works as expected, causing PufferPanel to chdir into the specified directory before reading/writing any files. However, it doesn't effect the `config.json` file, which appears to always be read from the working directory where PufferPanel is _started_, regardless of the `workDir` flag value.

The reason appears to be that PufferPanel [sets up the config file search path](https://github.com/PufferPanel/PufferPanel/blob/a226de68d69c02da2f81c2b10d8f1d3c49a150e1/config/config.go#L33) in an `init` function _before_ the `workDir` flag is processed. At first I was confused since it looked like `viper.AddConfigPath(".")` should make Viper look in the current directory at the time `ReadInConfig` is called; however, it seems that Viper actually [normalizes to an absolute path](https://github.com/spf13/viper/blob/97664ba0202811082423479584cc9542c702d69a/viper.go#L565) as soon as `AddConfigPath` is called. So the path that gets saved is PufferPanel's initial working directory, not the one specified with `workDir`.

This causes problems for the Windows service since it [starts in an inconvenient working directory](https://stackoverflow.com/a/884991/152208) with no apparent way to change it.

I think the `AddConfigPath` calls should be moved till after command-line parsing and `workDir` handling happens. This should preserve the same behavior for folks who don't specify `workDir`, while allowing those who do set it to put their config files into their preferred directory instead of the system's default.

**Comment by LordRalex** [comment]
I swore I tested that the config was in a sane spot when I did this on Windows services. Can you confirm it's actually in the system32 dir?

**Comment by bcat** [comment]
So I haven't had a chance to set up a Go dev environment at home yet to build the service myself with some extra logging (maybe this weekend?). But yes, it really does seem the service has an initial working directory of `C:\Windows\System32`. Here's my setup:

1. I installed PufferPanel 2.4.0 from the Windows service installer.
2. I created a `C:\ProgramData\PufferPanel` directory to hold my PufferPanel data.
3. I edited the service with `sc config` to have a binary path of `"C:\Program Files\PufferPanel\pufferpanel.exe" runService --workDir="C:\ProgramData\PufferPanel"`.
4. I dropped the following config into `C:\ProgramData\PufferPanel\config.json`:

```
{
  "panel": {
    "email": {
      "templates": "C:\\Program Files\\PufferPanel\\email\\emails.json"
    },
    "web": {
      "files": "C:\\Program Files\\PufferPanel\\www"
    }
  }
}
```

5. When I start the service, the config is ignored and startup fails (apparently because it's using the default relative email template path of `email/emails.json`, but my email templates aren't in `C:\ProgramData\PufferPanel`, they're in `C:\Program Files\PufferPanel` next to the excutable.
6. But if I copy `config.json` to `C:\Windows\System32\config.json` and try starting the service again, it starts up just fine, using the `email` and `www` paths I specified in the config.

So yes, it seems that:

* The service starts with a working directory of `C:\Windows\System32`.
* Additionally, the config file is searched for in the _initial_ working directory, not the `workDir` directory.
